modules/RTC/TraceablePeerConnection.js

```diff
@@ -1123,117 +1130,80 @@ TraceablePeerConnection.prototype._removeRemoteTrack = function(toBeRemoved) {
 };
 
 /**
- * Returns a map with keys msid/mediaType and <tt>TrackSSRCInfo</tt> values.
- * @param {RTCSessionDescription} desc the local description.
- * @return {Map<string,TrackSSRCInfo>}
+ * Processes the local SDP and creates an SSRC map for every local track.
+ *
+ * @param {string} localSDP - SDP from the local description.
+ * @returns {void}
  */
-TraceablePeerConnection.prototype._extractSSRCMap = function(desc) {
-    /**
-     * Track SSRC infos mapped by stream ID (msid) or mediaType (unified-plan)
-     * @type {Map<string,TrackSSRCInfo>}
-     */
+TraceablePeerConnection.prototype._processAndExtractSourceInfo = function(localSDP) {
     const ssrcMap = new Map();
 
-    /**
-     * Groups mapped by primary SSRC number
-     * @type {Map<number,Array<SSRCGroupInfo>>}
-     */
-    const groupsMap = new Map();
-
-    if (typeof desc !== 'object' || desc === null
-        || typeof desc.sdp !== 'string') {
-        logger.warn('An empty description was passed as an argument');
-
+    if (!localSDP || typeof localSDP !== 'string') {
```

Comment:
Is this an error? Should a warning be logged?

Reply:
This technically wouldn't happen since the browser will always produce an  SDP for createOffer/createAnswer. The call itself will fail and user kicked out when that happens so I thought it's ok not to display the error here.

Reply:
In that case, let's throw an exception, as if it was an assertion. If the invariant is ever broken, better do it as loud as possible.